### PR TITLE
Milestone3

### PR DESCRIPTION
**Pull request by guanxiaotongyz**
1. insert translation line 
2. all bottom just click once 
3. can download with html 
4. can revise multiple time

### CODE REVIEW

File: src/app/demo/demo.component.ts [hunk 1]
```diff
@@ -27,7 +28,69 @@ export class DemoComponent implements OnInit {
...
+      innerbutton.addEventListener("click", () => {
+        sentence.insertAdjacentHTML(
+          "beforeend",
+          '<br><span class = "translation" contenteditable = True>Translation</span>'
+        );
+        innerbutton.disabled = true;
+      });
+      sentence.insertAdjacentElement("afterend", innerbutton);
+    });
+  }
```
Comment (yoxchen95) [inline src/app/demo/demo.component.ts:51]: This function is good to insert translationline for the pages. Excellent.

Comment (yoxchen95) [reply src/app/demo/demo.component.ts:51]: And the once button function also inspires me for the milestone2.

File: src/app/demo/demo.component.ts [hunk 2]
```diff
@@ -27,7 +28,69 @@ export class DemoComponent implements OnInit {
...
+      count++;
+    });
+
+    const serializer = new XMLSerializer();
+    const xmlStr = serializer.serializeToString(doc);
+    this.b64Inputs[1] =
+      this.b64Inputs[1].slice(0, this.b64Inputs[1].indexOf(",") + 1) +
+      this.b64Service.utf8_to_b64(xmlStr);
+  }
+
```
Comment (yoxchen95) [inline src/app/demo/demo.component.ts:89]: It is clever to use p tag to insert the new translation line into xml file and add it to the download file.It is good

**Review by yoxchen95 — APPROVED** [review]

**Review by crystaljesica — APPROVED** [review]
after figuring out the bug of the test, it can pass the cicd pipeline. Looks good to me. Ready to merge.

**Pull request merged**